source/kraken/realtime.cpp

```diff
@@ -332,14 +339,16 @@ create_disruption(const std::string& id,
                 }
                 // we update the trip status if the stoptime status is the most important status
                 // the most important status is DELAYED then DELETED
-                most_important_stoptime_status = std::max(most_important_stoptime_status, departure_status);
-                most_important_stoptime_status = std::max(most_important_stoptime_status, arrival_status);
+                most_important_stoptime_status = std::max({most_important_stoptime_status,
+                    departure_status ,arrival_status});
 
                 StopTimeUpdate st_update{std::move(stop_time), message, departure_status, arrival_status};
                 impact->aux_info.stop_times.emplace_back(std::move(st_update));
             }
 
             switch (most_important_stoptime_status) {
+            case StopTimeUpdate::Status::ADDED:
```

Comment:
It would have been nice to extract this in a separate function as it's already super loooooooooong.